staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -37,12 +37,34 @@ var ForeverTestTimeout = time.Second * 30
 // NeverStop may be passed to Until to make it never stop.
 var NeverStop <-chan struct{} = make(chan struct{})
 
-// StartUntil starts f in a new goroutine and calls done once f has finished.
-func StartUntil(stopCh <-chan struct{}, wg *sync.WaitGroup, f func(stopCh <-chan struct{})) {
+// Group is an interface to decouple code from sync.WaitGroup.
+type Group interface {
+	Add(delta int)
+	Done()
+}
+
+// StartWithinGroupUntil adds 1 to the group, starts f in a new goroutine and calls wg.Done once f has finished.
+// stopCh is passed to f as an argument.
+func StartWithinGroupUntil(stopCh <-chan struct{}, wg Group, f func(stopCh <-chan struct{})) {
```

Comment:
I like both the signature of the function and the documentation, but it builds in an assumption as to _why_ the function needs this channel as a parameter, and what it means when the channel is available. At present, the function doesn't pay attention to the channel. The "until" qualifier is about the assumed behavior of the "f" parameter, but we need not and cannot make such an assumption. Consider generalizing it with a slight change:
```go
func StartWithChannelWithinGroup(ch <-chan struct[}, wg Group, f func(ch <-chan struct{}))
```
If that name makes sense, consider renaming its sibling to `StartWithContextWithinGroup`. The names are longer, but I think their meaning is clearer.